### PR TITLE
TSDB: Use t.Cleanup to delete temporary files

### PR DESCRIPTION
**Pull request by roidelapluie**
cc @krasi-georgiev @codesome 

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: tsdb/compact_test.go [hunk 1]
```diff
@@ -900,10 +900,9 @@ func BenchmarkCompactionFromHead(b *testing.B) {
 // This is needed for unit tests that rely on
 // checking state before and after a compaction.
 func TestDisableAutoCompactions(t *testing.T) {
-	db, closeFn := openTestDB(t, nil, nil)
+	db := openTestDB(t, nil, nil)
 	defer func() {
```
Comment (bwplotka) [inline tsdb/compact_test.go:904]: `t.Cleanup(func() {` as well?

Comment (roidelapluie) [reply tsdb/compact_test.go:904]: Here there is no real added value, but I can do it also.

Comment (roidelapluie) [reply tsdb/compact_test.go:904]: I'd rather limit the scope at this for this PR, and look for other places where it would make sense.

Comment (bwplotka) [reply tsdb/compact_test.go:904]: :+1:

**Comment by roidelapluie** [comment]
I have rebased on top if #7625 ; will merge on green

File: tsdb/db_test.go [hunk 2]
```diff
@@ -59,10 +59,11 @@ func openTestDB(t testing.TB, opts *Options, rngs []int64) (db *DB, close func()
 	}
 	testutil.Ok(t, err)
 
-	// Do not close the test database by default as it will deadlock on test failures.
-	return db, func() {
+	// Do not Close() the test database by default as it will deadlock on test failures.
+	t.Cleanup(func() {
```
[the remark this answers is not in the document]
Comment (roidelapluie) [reply tsdb/db_test.go:69]: > This is my favourite use-case for `t.Cleanup`!

go 1.15 will have directly t.Tempdir:
https://tip.golang.org/pkg/testing/#T.TempDir

**Pull request merged**